internal/goro/goro.go

```diff
@@ -40,15 +40,23 @@ type (
 	}
 )
 
-// Go launches the supplied function in its own goroutine and returns back a
-// *Handle that serves as a handle to the goroutine itself.
-func Go(ctx context.Context, f func(context.Context) error) *Handle {
+// NewHandle creates a *Handle that serves as a handle to a goroutine. The
+// caller should call Go exactly once on the returned value. NewHandle and Go
+// are separate function so that the *Handle can be stored into a field before
+// the goroutine starts, which makes it possible for the goroutine to call
```

Comment:
Just curious and sorry for the drive-by but did goro.Group not work for this purpose?

Reply:
Comments are always welcome, especially when it's your code.

I looked at Group but it was missing a Done() channel. I guess I could have exposed the context's Done channel? At that point it seems like we should just unify Goro and Group since they're so similar.